shared/models/videos/video-resolution.enum.ts

```diff
@@ -1,7 +1,59 @@
+import { VIDEO_TRANSCODING_FPS } from '../../../server/initializers'
+
 export enum VideoResolution {
   H_240P = 240,
   H_360P = 360,
   H_480P = 480,
   H_720P = 720,
   H_1080P = 1080
 }
+
+/**
+ * Sources for individual quality levels:
+ * Google Live Encoder: https://support.google.com/youtube/answer/2853702?hl=en
+ * YouTube Video Info (tested with random music video): https://www.h3xed.com/blogmedia/youtube-info.php
+ */
+export function getTargetBitrate (resolution: VideoResolution, fps: number) {
+  switch (resolution) {
+  case VideoResolution.H_240P:
+    // quality according to Google Live Encoder: 400 - 1,000 Kbps
+    // Quality according to YouTube Video Info: 186 Kbps
+    return 250 * 1000
+  case VideoResolution.H_360P:
+    // quality according to Google Live Encoder: 400 - 1,000 Kbps
+    // Quality according to YouTube Video Info: 480 Kbps
+    return 500 * 1000
```

Comment:
If this is the max rate, I think we should use the max youtube rate so 1000 Kbps no (even if we `*1.5` below)? (and it's higher on vimeo for example)

Reply:
I really wasn't sure about the numbers, and I couldn't find any great sources either. I tried to keep close to the actual bitrates Youtube uses, because they spent a lot of time optimizing it. If the bitrates are too high, it won't work well for people with slow internet (and they will keep using youtube).

Reply:
Yes but we talk about the max bitrate here, not the overall bitrate. Maybe we should use `-b:v` ffmpeg options instead of `maxrate`? I'm afraid with `maxrate`, we slaughter certain sequences of the video when it finally has an acceptable overall bitrate

Reply:
The site I linked in the code [1] specifically mentions using maxrate and bufsize for streaming, as does the FFmpeg documentation [2]. Unless you have a different source that recommends b:v for streaming, I think we should go with this.

[1] https://slhck.info/video/2017/03/01/rate-control.html
[2] https://trac.ffmpeg.org/wiki/EncodingForStreamingSites